### PR TITLE
refactor: component migration

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
refactor: move components from dashboard to fe-common-lib

Fixes https://github.com/devtron-labs/devtron/issues/4907

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/common/formFields/ButtonWithLoader.tsx [hunk 1]
```diff
@@ -12,6 +12,10 @@ export interface ButtonProps {
     type?: 'submit' | 'reset' | 'button'
 }
 
+/**
+ * @deprecated Use `ButtonWithLoader` from common lib instead
+ */
+
```
Comment (eshankvaish) [inline src/components/common/formFields/ButtonWithLoader.tsx:18]: .

File: src/components/app/list-new/AppList.tsx [hunk 2]
```diff
@@ -21,15 +22,14 @@ import DevtronAppListContainer from '../list/DevtronAppListContainer'
...
-import { mainContext } from '../../common/navigation/NavigationRoutes'
 import '../list/list.scss'
 import EAEmptyState, { EAEmptyStateType } from '../../common/eaEmptyState/EAEmptyState'
 import ExportToCsv from '../../common/ExportToCsv/ExportToCsv'
 import { FILE_NAMES } from '../../common/ExportToCsv/constants'
 import { getAppList } from '../service'
 import { getUserRole } from '../../../Pages/GlobalConfigurations/Authorization/authorization.service'
 import { APP_LIST_HEADERS, StatusConstants } from './Constants'
-import HeaderWithCreateButton from '../../common/header/HeaderWithCreateButton/HeaderWithCreateButton'
+import HeaderWithCreateButton from '@devtron-labs/devtron-fe-common-lib'
```
Comment (eshankvaish) [inline src/components/app/list-new/AppList.tsx:32]: .

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/Jobs/JobList/JobsList.tsx [hunk 3]
```diff
@@ -7,11 +7,12 @@ import {
     ServerErrors,
     useAsync,
     DevtronProgressing,
+    useMainContext,
 } from '@devtron-labs/devtron-fe-common-lib'
 import * as queryString from 'query-string'
 import { URLS } from '../../../config'
 import { Filter, FilterOption } from '../../common'
-import HeaderWithCreateButton from '../../common/header/HeaderWithCreateButton/HeaderWithCreateButton'
+import HeaderWithCreateButton from '@devtron-labs/devtron-fe-common-lib'
```
[the remark this answers is not in the document]
Comment (eshankvaish) [reply src/components/Jobs/JobList/JobsList.tsx:15]: bump common lib version

**Review by vivek-devtron — APPROVED** [review]

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]
Please remove sonar cloud issues and btw, try to have external imports as first one

**Review by AbhishekA1509 — APPROVED** [review]

**Review by eshankvaish — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**